src/core/dev/hw_queue_tx.h

```diff
@@ -149,10 +162,15 @@ class hw_queue_tx : public xlio_ti_owner {
 
     void reset_inflight_zc_buffers_ctx(void *ctx);
 
-    void credits_return(unsigned credits) { m_sq_free_credits += credits; }
+    void credits_return(unsigned credits)
```

Comment:
I believe we need those for dpcp path. I beleive you need to differentiate between doca and dpcp .

Reply:
Done.